### PR TITLE
Update Developer's Guide for Logic, Storage and DateTime Implementation

### PR DESCRIPTION
**Pull request by MerlinLim**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
...
+The components follow the general sequence to execute a command:
+
+1. `Logic` uses the `XYZBookParser` class to parse the user command.
+1. This results in a `XYZCommand` object which is executed by the `LogicXYZManager`.
 1. The command execution can affect the `Model` (e.g. adding a person).
 1. The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
 1. In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
 
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+Given below is the Sequence Diagram for interactions within the `Logic` component for API call of any command.
```
Comment (claraadora) [inline docs/DeveloperGuide.md:122]: sequence diagram

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
...
+1. This results in a `XYZCommand` object which is executed by the `LogicXYZManager`.
 1. The command execution can affect the `Model` (e.g. adding a person).
 1. The result of the command execution is encapsulated as a `CommandResult` object which is passed back to the `Ui`.
 1. In addition, the `CommandResult` object can also instruct the `Ui` to perform certain actions, such as displaying help to the user.
 
-Given below is the Sequence Diagram for interactions within the `Logic` component for the `execute("delete 1")` API call.
+Given below is the Sequence Diagram for interactions within the `Logic` component for API call of any command. 
 
-![Interactions Inside the Logic Component for the `delete 1` Command](images/DeleteSequenceDiagram.png)
+![Interactions Inside the Logic Component for any command](images/CommandSequenceDiagram.png)
```
Comment (claraadora) [inline docs/DeveloperGuide.md:124]: Remove the color for outer box 
Screenshot 2020-11-09 at 4 02 06 PM

Comment (claraadora) [reply docs/DeveloperGuide.md:124]: - Add XYZ note 
- I think execute("XYZ args") shouldnt be XYZ args. If any command then can just put execute(command).
- there is a d in d:XYZCommand --> not sure what is it and if its meaningful
- i think the condition in opt can be "command affects models"

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
...
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+<div markdown="span" class="alert alert-info">:information_source: 
+**Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X)
```
Comment (claraadora) [inline docs/DeveloperGuide.md:127]: There's no DeleteCommandParser in the diagram

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
...
-![Interactions Inside the Logic Component for the `delete 1` Command](images/DeleteSequenceDiagram.png)
+![Interactions Inside the Logic Component for any command](images/CommandSequenceDiagram.png)
 
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+<div markdown="span" class="alert alert-info">:information_source: 
+**Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X) 
+but due to a limitation of PlantUML, the lifeline reaches the end of diagram.<br>
+</div>
+<div markdown="span" class="alert alert-primary">:bulb:
+**Tip**: Refer to [Switch Mode Feature](#switch-mode-feature) below for more details on the Switch Mode feature.
```
Comment (claraadora) [inline docs/DeveloperGuide.md:131]: Im not sure but I think the Logic component section doesn't mention switch command? So this tip might be coming out of nowhere

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -132,11 +155,36 @@ The `Model`:
 
 ![Structure of the Storage Component](images/StorageClassDiagram.png)
 
-**API** : [`Storage.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/storagePerson/Storage.java)
+<div markdown="span" class="alert alert-info"> 
+**:information_source: Note:** ABC refers to `Deliverable`, `Meeting` and `Person`.
```
Comment (claraadora) [inline docs/DeveloperGuide.md:159]: Same for this

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -132,11 +155,36 @@ The `Model`:
 
 ![Structure of the Storage Component](images/StorageClassDiagram.png)
```
Comment (claraadora) [inline docs/DeveloperGuide.md:156]: Label XYZ on the diagram

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -132,11 +155,36 @@ The `Model`:
...
+**:information_source: Note:** ABC refers to `Deliverable`, `Meeting` and `Person`.
+
+</div>
+
+**API** : 
+[`StorageDeliverable.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/storage/deliverable/StorageDeliverable.java)
+[`StorageMeeting.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/storage/meeting/StorageMeeting.java)
+[`StoragePerson.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/storage/person/StoragePerson.java)
 
 The `Storage` component:
```
Comment (claraadora) [inline docs/DeveloperGuide.md:168]: I think no `` formatting for `Storage` component
Need to standardize this too @chrystalquek @gabztcr @shadowezz

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -132,11 +155,36 @@ The `Model`:
...
 * can save `UserPref` objects in json format and read it back.
-* can save the address book data in json format and read it back.
+* can save the data in json format and read it back.
+
+For saving files, storage follows this sequence:
+
+1. When `ABCBook` is updated, `StorageABCManagers` saves the newly updated book.
+1. The newly updated book is passed to `JsonSerliazableABCBook`.
+1. Each item in `ABCBook` is serialized by `JsonAdaptedABC` before overwriting the current Jsonfile
```
Comment (claraadora) [inline docs/DeveloperGuide.md:176]: XYZ
Is JsonFile a class/code? If yes need `` formatting. If not, maybe change to "current json file"

File: docs/DeveloperGuide.md [hunk 9]
```diff
@@ -132,11 +155,36 @@ The `Model`:
...
+
+Given below is the Sequence Diagram for data being stored.
+
+![Interactions Inside the Storage Component for Any Book](images/SaveStorageSequenceDiagram.png)
```
Comment (claraadora) [inline docs/DeveloperGuide.md:180]: - ABC to XYZ (label + note) 
- Lighter color

File: docs/DeveloperGuide.md [hunk 10]
```diff
@@ -132,11 +155,36 @@ The `Model`:
...
+
+For reading files, based on the `UserPrefs` provided, storage will find the Json Files and load the data from there. 
+
+<div markdown="span" class="alert alert-info">:information_source: 
+**Note:** 
+If the Json files are missing or if the path is missing, a new set of Json files will be generated with a set of sample items.
```
Comment (claraadora) [inline docs/DeveloperGuide.md:186]: Based on this I think then the above should be Json files not Json Files?

File: docs/DeveloperGuide.md [hunk 11]
```diff
@@ -147,38 +195,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 ## **Implementation**
 
 This section describes some noteworthy details on how certain features are implemented.
-### [In Progress] \[DateTime\]
+### [Proposed] \[Date and Time verfication\]
```
Comment (claraadora) [inline docs/DeveloperGuide.md:198]: Delete Proposed

File: docs/DeveloperGuide.md [hunk 12]
```diff
@@ -147,38 +195,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
 
 #### Proposed Implementation
-The implementation allows users to parse and compare unique DateTime types. 
+The implementation allows users to parse and compare unique Date and Time types.
```
Comment (claraadora) [inline docs/DeveloperGuide.md:201]: `DateTime` and `Time` objects maybe?

File: docs/DeveloperGuide.md [hunk 13]
```diff
@@ -147,38 +195,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
 
+![Structure of DateTime and Time implementations](images/DateTimeClass.png)
+
+**Time type**:
+To parse, Time should be in the following format: **`HH:mm`**
```
Comment (claraadora) [inline docs/DeveloperGuide.md:206]: `Time`

File: docs/DeveloperGuide.md [hunk 14]
```diff
@@ -147,38 +195,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+**Time type**:
+To parse, Time should be in the following format: **`HH:mm`** 
+* Single digits fields must include leading zero: `01:10`.
+
+Time will throw a parsing error if
+* `00:00:59` Format is wrong (e.g missing or additional digit).
+* `24:00` Invalid range (e.g invalid leap year).
+
+**DateTime type**:
 To parse, DateTime should be in the following format: **`dd-MM-yyyy HH:mm`**
```
Comment (claraadora) [inline docs/DeveloperGuide.md:214]: `Time` and `DateTime`

File: docs/DeveloperGuide.md [hunk 15]
```diff
@@ -147,38 +195,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+**DateTime type**:
 To parse, DateTime should be in the following format: **`dd-MM-yyyy HH:mm`** 
 * Single digits fields must include leading zero: `01-01-0101 01:10`.
-* Valid Calendar Range: \[`01-01-0001 00:00` - `31-12-9999 23:59`\].
+* Valid Calendar Range: \[`01-01-2019 00:00` - `31-12-9999 23:59`\].
 
 DateTime will throw a parsing error if
 * `1-10-2020 00:00:59` Format is wrong (e.g missing or additional digit).
 * `31-02-2020 00:00` Invalid range (e.g invalid leap year).
```
Comment (claraadora) [inline docs/DeveloperGuide.md:220]: I thinkt he example should come after the situation i.e. Format is wrong (e.g. 1-10 etc...)

File: docs/DeveloperGuide.md [hunk 16]
```diff
@@ -147,38 +195,45 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
-
-![DateTimeClassDiagram](images/DateTimeClass.png)
-* DateTime is a class that can be used by all models.
-* From, To and Deadline are fields which extend from DateTime.
-
 DateTime can be used to compare with other DateTime objects:
-* Enable deliverables to be sorted based on which one is due the earliest.
+* Enable deliverables or meetings to be sorted based on which one is due the earliest.
+*Refer to [Autosort feature](#proposed-autosort-feature) to view this implementation.*
+* Used to identify invalid inputs (e.g datetime in `from` is after `to`).
```
Comment (claraadora) [inline docs/DeveloperGuide.md:225]: datetime might be incorrect

**Review by claraadora — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 17]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
 
 ![Structure of the Logic Component](images/LogicClassDiagram.png)
```
[the remark this answers is not in the document]
Comment (MerlinLim) [reply docs/DeveloperGuide.md:93]: Have checked and included LogicDispatcher

File: docs/DeveloperGuide.md [hunk 18]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
...
 
-**API** :
-[`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
+<div markdown="block" class="alert alert-info"> 
+:information_source: **Note:** ABC refers to `Deliverable`, `Meeting`, `Person`. 
+XYZ refers to any command call, e.g Add, find, delete, etc.
```
[the remark this answers is not in the document]
Comment (MerlinLim) [reply docs/DeveloperGuide.md:97]: Will update

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:159]: Removed

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:131]: Initially I had mentioned it in the description but removed it. Wil change accordingly

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:127]: Amended

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:124]: XYZ is meant to represent the the type of command
d is meant to represent the object, will change to `abcCommand` instead
Will amend accordingly

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:156]: Amended

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:168]: Removed accordingly

File: docs/DeveloperGuide.md [hunk 19]
```diff
@@ -92,20 +92,43 @@ The `UI` component:
...
+
+**API** : <br>
+* [`LogicDeliverable.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/logic/LogicDeliverable.java)
+* [`LogicMeeting.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/logic/LogicMeeting.java)
+* [`LogicPerson.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/logic/LogicPerson.java)
+* [`LogicDispatcher.java`](https://github.com/AY2021S1-CS2103T-F11-2/tp/tree/master/src/main/java/seedu/address/logic/LogicDispatcher.java)
+
+The `Logic` component parses the user commands and executes them. There are different pa`LogicDispatcher` selects the correct parser to 
+
+This is the list of what `Model` components affected by their respective `Logic` components:
```
[the remark this answers is not in the document]
Comment (MerlinLim) [reply docs/DeveloperGuide.md:109]: removed accordingly

File: docs/DeveloperGuide.md (hunk 8 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:176]: All mentions of XYZ is a class/code

File: docs/DeveloperGuide.md (hunk 9 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:180]: Amended

File: docs/DeveloperGuide.md (hunk 10 above)
Comment (MerlinLim) [reply docs/DeveloperGuide.md:186]: standardized all to be lowercase

**Review by claraadora — APPROVED** [review]
LGTM

**Pull request merged**